src/lib.rs

```diff
@@ -118,6 +120,35 @@ impl Library {
         imp::Library::new(filename).map(From::from)
     }
 
+    fn resolve_name<'a>(name: &'a str) -> Cow<'a, str> {
+        match (name.starts_with(DLL_PREFIX), name.ends_with(DLL_SUFFIX)) {
+            (true, true) => Cow::Borrowed(name),
+            (true, false) => Cow::Owned(format!("{}{}", name, DLL_SUFFIX)),
+            (false, true) => Cow::Owned(format!("{}{}", DLL_PREFIX, name)),
+            (false, false) => Cow::Owned(format!("{}{}{}", DLL_PREFIX, name, DLL_SUFFIX)),
+        }
+    }
+
+    /// Loads a library and does resolve it's name to match platform-specific
+    /// naming schemes.
+    ///
+    /// For a given library called `engine`, this method will resolve the name
+    /// to the following:
+    /// - `Linux`: `libengine.so`
+    /// - `macOS`: `libengine.dylib`
```

Comment:
Given that loadable modules (`MH_BUNDLE`) on macOS are commonly `.so` and conventionally `.bundle` does `dylib` actually make sense? Should we also try the other two options before finally failing?

Reply:
I am not really familiar with the macOS specifics and have to read up on that part. But seeing that the stdlib of Rust itself uses `.dylib` as the extension I'd probably opt for only checking that file. Even though (after a really quick! read), it seems like `.bundle` files are often used as well. But that seems "too" platform-specific and might be better of handled by the caller using the older method `new`. 



> Can `MH_DYLIB` DSOs be loaded by `dlopen` at all?

Yes, they can definitely be loaded using `dlopen` (first paragraph in [description](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/dlopen.3.html))